app/services/foreman_templates/template_importer.rb

```diff
@@ -0,0 +1,158 @@
+                   metadata['model'].constantize.import!(name, text, metadata)
+                 else
+                   # For backwards-compat before "model" metadata was added
+                   case metadata['kind']
+                   when 'ptable'
+                     Ptable.import!(name, text, metadata)
+                   when 'job_template'
+                     # TODO: update REX templates to have `model` and delete this
+                     update_job_template(name, text)
+                   else
+                     ProvisioningTemplate.import!(name, text, metadata)
+                   end
+                 end
+
+          if data[:status] == true && @verbose
+            puts data[:result]
+            puts data[:diff]
+          elsif data[:status] == false
+            puts "Template \"#{name}\": #{data[:result]}"
+          end
+        rescue NoKindError
+          puts "  Error: Unknown template type '#{metadata['kind']}'"
+          next
+        end
+      end
+    end
+
+    def get_default_branch(repo)
+      branch_names = repo.branches.map(&:name).uniq
+
+      # Always use develop on Foreman-nightly, if present, or else relevant stable branch
+      target = SETTINGS[:version].tag == 'develop' ? 'develop' : "#{SETTINGS[:version].short}-stable"
+      return target if branch_names.include?(target)
+
+      # stay on default branch as fallback
+      nil
+    end
+
+    def parse_metadata(text)
+      # Pull out the first erb comment only - /m is for a multiline regex
+      extracted = text.match(/<%\#(.+?).-?%>/m)
+      extracted.nil? ? {} : YAML.load(extracted[1])
+    end
+
+    def update_job_template(name, text)
+      return {
+        :status => false,
+        :result => 'Skipping job template import, remote execution plugin is not installed.'
+      } unless defined?(JobTemplate)
+      template = JobTemplate.import(
+        text.sub(/^name: .*$/, "name: #{name}"),
+        :update => true
+      )
+
+      c_or_u = template.new_record? ? 'Created' : 'Updated'
+      id_string = ('id' + template.id) rescue ''
+
+      if template.template != template.template_was
+        diff = Diffy::Diff.new(
+          template.template_was,
+          template.template,
+          :include_diff_info => true
+        ).to_s(:color)
+      end
+
+      result = "  #{c_or_u} Template #{id_string}:#{name}"
+      { :diff => diff, :status => template.save, :result => result }
+    end
+
+    def purge!
+      # Available options:
+      #* verbose => Print extra information during the run [false]
+      #* not     => negate prefix / filter
+      #* kind    => Template kind
+      #* prefix  => The string all imported templates should begin with [Community ]
+      #* filter  => Import names matching this regex (case-insensitive; snippets are not filtered)
+
+      @verbose = ( ENV['verbose'] and ENV['verbose'] != 'false' ) ? true : false
+      match    = ENV['not'] ? 'NOT LIKE' : 'LIKE'
```

Comment:
I'll have a look, I figured I'd use it to negate a search might not have used it yet...